### ISSUE TITLE
Server Crashes upon loading/generating the end

### ISSUE THREAD

**Issue by pandaperson5592** [AllTheMods/SLOP2-slice-of-pie-2#19]
i have no idea if its just me but when me and my friend jumped into the end portal the server crashed. After restarting the server it would crash before either of us would be able to play and when a different person joined and they would be in the overworld the server would be just fine until one of us who were in the end joined. i tried deleteing the world folder of the end and that didnt change anything with a new end. We were able to save our players by TPing the people who were in the end to the overworld and then it worked fine Maybe like some weird Terra Generation thing?

[crash-2021-08-17_20.08.06-server.txt](https://github.com/AllTheMods/SLOP2-slice-of-pie-2/files/7004277/crash-2021-08-17_20.08.06-server.txt)

[crash-2021-08-17_20.21.46-server.txt](https://github.com/AllTheMods/SLOP2-slice-of-pie-2/files/7004278/crash-2021-08-17_20.21.46-server.txt)

**Comment by MadWomen** [comment]
this is a common issue with repurposed structures and will be fixed in the next version, to fix this issue now, you can update repurposed structures manually on both ends.

**Comment by TelepathicGrunt** [comment]
Yeah sorry about that. There’s was several pieces of methods that were clientside only when similar methods were serverside safe and I missed that. Like getStepX() is safe in direction but step() isn’t. Then after I fixed that, turns out Vextor3f.mul() is also not serverside safe. 

Newest version of RS is safe tho. I ran it on server to triple check and make sure nothing blew up. But let me know if there’s any further issues
